From d786da67470dd879c5bbe13d38a7f95f2b598626 Mon Sep 17 00:00:00 2001
From: Simone Mainardi <mainardi@ntop.org>
Date: Fri, 5 Aug 2016 17:53:34 +0200
Subject: [PATCH] Security Fix: users enumeration

A user reported us

"""
I would like to report a possible security issue in ntopng.

When connected with a regular, unprivileged user, it is possible to force-navigate to the URL: http://192.168.2.176:3000/lua/admin/validate_new_user.lua?user=<anyuser>&networks=0.0.0.0/0,::/0
Due to this, it is possible to replace <anyuser> with a list of users and enumerate the users that exist in the system.
"""

This commit fixes the issue
---
 scripts/lua/modules/lua_utils.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/lua/modules/lua_utils.lua b/scripts/lua/modules/lua_utils.lua
index 3e5e36b..8ef739e 100644
--- a/scripts/lua/modules/lua_utils.lua
+++ b/scripts/lua/modules/lua_utils.lua
@@ -1781,7 +1781,7 @@ end
  -- ##############################################
 
 function haveAdminPrivileges()
-   if(isAdministrator) then
+   if(isAdministrator()) then
       return(true)
    else
       ntop.dumpFile(dirs.installdir .. "/httpdocs/inc/header.inc")
